src/plugins/console/public/lib/autocomplete/engine.js

```diff
@@ -127,7 +127,10 @@ export function populateContext(tokenPath, context, editor, includeAutoComplete,
           if (!_.isObject(term)) {
             term = { name: term };
           }
-          autoCompleteSet.push(term);
+          // Make sure we don't have duplicates
+          if (!_.find(autoCompleteSet, { name: term.name })) {
```

Comment:
I wonder if it would make more sense to use a `Set` here so that items can be added without needing to loop over the array. Does the array get big? Do we have any performance concerns with this code?

Reply:
Yeah I think we can reduce complexity if we avoid the find, I ended up going for a map rather than a set to avoid having to serialize the `term` with JSON.stringify. It doesnt get big at all afaict, mostly autocomplete terms are in the order of <=20 or so, so should be pretty safe performance wise